src/components/contextless-android-activity-toggle/contextless-android-activity-toggle.js

```diff
@@ -23,7 +23,19 @@ export default class ContextlessAndroidActivityToggle extends React.Component {
 
     let selectedCode = '';
     if (context) {
-      selectedCode = this.checkPreference('kotlin') ? kotlin : java;
+      if (kotlin === undefined) {
+        /* If there is no kotlin code, use java. */
+        selectedCode = java;
+      } else if (java === undefined) {
+        /* If there is no java code, use kotlin. */
+        selectedCode = kotlin;
+      } else if (this.checkPreference('kotlin')) {
+        /** If there is both java and kotlin code,
+         * use the preferred language. */
+        selectedCode = kotlin;
+      } else {
+        selectedCode = java;
+      }
```

Comment:
@colleenmcginnis you may want to consider breaking this out into a function that you can test. Nested logic can be hard to parse and having test cases can help debug and future problems (if any). If you'd like to give this a try, here are some suggestions on how to approach it:

1. Above `export default class ContextlessAndroidActivityToggle..` you could create a new function, something like `export function findSelectedCode(kotlin, java, preference) {}` (this is passing the kotlin code, java code, and the user's preference).
2. The interior of the new function could be lines 24 + 26 - 38, and the function would return the value of `selectedCode`. The only other change you'd want to make is at line 32, where you could evaluate `preference === 'kotlin'` instead of invoking `this.checkPreference` since we are now outside of the class and cannot reach that function.
3. Coming back to line 26, we can call that function: `selectedCode = findSelectedCode(kotlin, java, context.preferredLanguage.android)`
4. Create a new test file in the component's `__tests__` dir, import the function `import {findSelectedCode} from '../contextless-android-activity-toggle';` and build out a series of test cases to make sure it returns the selected code. Example: `expect(findSelectedCode('kotlin code', 'java code', 'kotlin')).toEqual("kotlin code");`